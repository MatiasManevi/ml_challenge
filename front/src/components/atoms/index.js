export { default as Logo } from 'components/atoms/Logo';
export { default as Price } from 'components/atoms/Price';
export { default as SearchInput } from 'components/atoms/SearchInput';
export { default as SearchButton } from 'components/atoms/SearchButton';
export { default as PrimaryButton } from 'components/atoms/PrimaryButton';
