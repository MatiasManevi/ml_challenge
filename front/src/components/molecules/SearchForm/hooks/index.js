export { default as useSearchForm } from 'components/molecules/SearchForm/hooks/useSearchForm';
