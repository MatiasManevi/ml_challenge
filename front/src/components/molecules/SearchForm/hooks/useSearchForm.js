import React from 'react';
import { useHistory } from 'react-router-dom';

const useSearchForm = () => {
	const [query, setQuery] = React.useState('');
	const history = useHistory();

	const handleInputChange = (e) => {
		setQuery(e.target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		history.push(`/items?search=${query}`);
		setQuery('');
	};

	return {
		handleInputChange,
		handleSubmit,
		query
	};
};

export default useSearchForm;
