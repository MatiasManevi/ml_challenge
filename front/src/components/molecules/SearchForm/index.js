export { default } from 'components/molecules/SearchForm/SearchForm';
