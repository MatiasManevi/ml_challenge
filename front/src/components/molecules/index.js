export { default as SearchForm } from 'components/molecules/SearchForm';
