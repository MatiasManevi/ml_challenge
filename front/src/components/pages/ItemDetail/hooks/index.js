export { default as useItemDetail } from 'components/pages/ItemDetail/hooks/useItemDetail';
