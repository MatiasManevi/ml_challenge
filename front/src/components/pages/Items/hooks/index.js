export { default as useItems } from 'components/pages/Items/hooks/useItems';
