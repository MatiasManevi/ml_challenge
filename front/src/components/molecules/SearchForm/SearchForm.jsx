import React from 'react';

import './SearchForm.scss';
import { SearchButton, SearchInput } from 'components/atoms';
import { useSearchForm } from 'components/molecules/SearchForm/hooks';

const SearchForm = () => {
	const { handleInputChange, handleSubmit, query } = useSearchForm();

	return (
		<form className="search-form">
			<SearchInput value={query} changeHandler={handleInputChange} />
			<SearchButton clickHandler={handleSubmit} />
		</form>
	);
};

export default SearchForm;
