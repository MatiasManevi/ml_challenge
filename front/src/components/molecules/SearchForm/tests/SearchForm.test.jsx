import React from 'react';
import { shallow, mount } from 'enzyme';

import { SearchButton } from 'components/atoms';
import SearchForm from '../SearchForm';
import * as hooks from '../hooks';

jest.mock('../hooks');

const handleSubmit = jest.fn();
const handleInputChange = jest.fn();
const stateSpy = jest.spyOn(hooks, 'useSearchForm');
const query = 'Laptop';

stateSpy.mockReturnValue({
	handleInputChange,
	handleSubmit,
	query
});

describe('SearchForm', () => {
	it('should match snapshot', () => {
		const wrapper = shallow(<SearchForm />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should submit a query', () => {
		const wrapper = mount(<SearchForm />);
		wrapper.find(SearchButton).simulate('click');
		expect(handleSubmit).toHaveBeenCalled();
	});
});
