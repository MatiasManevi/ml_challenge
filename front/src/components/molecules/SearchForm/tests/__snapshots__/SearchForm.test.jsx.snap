// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot 1`] = `
<form
  className="search-form"
>
  <SearchInput
    changeHandler={[MockFunction]}
    value="Laptop"
  />
  <SearchButton
    clickHandler={[MockFunction]}
  />
</form>
`;
